core/lib/types/src/pubdata_da.rs

```diff
@@ -1,21 +1,42 @@
+use chrono::{DateTime, Utc};
 use num_enum::TryFromPrimitive;
 use serde::{Deserialize, Serialize};
+use zksync_basic_types::L1BatchNumber;
 use zksync_config::configs::eth_sender::PubdataSendingMode;
 
 /// Enum holding the current values used for DA Layers.
 #[repr(u8)]
 #[derive(Debug, Clone, Copy, Deserialize, PartialEq, Serialize)]
 #[derive(TryFromPrimitive)]
 pub enum PubdataDA {
+    /// Pubdata is sent to the L1 as a tx calldata.
     Calldata = 0,
+    /// Pubdata is sent to L1 as EIP-4844 blobs.
     Blobs,
+    /// Pubdata is sent to the external storage (GCS/DA layers) or not sent at all.
+    Custom,
 }
 
 impl From<PubdataSendingMode> for PubdataDA {
     fn from(value: PubdataSendingMode) -> Self {
         match value {
             PubdataSendingMode::Calldata => PubdataDA::Calldata,
             PubdataSendingMode::Blobs => PubdataDA::Blobs,
+            PubdataSendingMode::Custom => PubdataDA::Custom,
         }
     }
 }
+
+/// Used as a wrapper for the pubdata to be stored in the GCS.
+pub struct StorablePubdata {
```

Comment:
This object seems to be used only by `default_da_clients` crate, but now it spans over three crates (`zksync_types`, `zksync_object_store`, `zksync_default_da_clients`). I think it's better to move both definition and trait implementation to the relevant module in the `default_da_crates`.

Reply:
Also nit: it misses `#[derive(Debug)]`.